system-int-test/test/cmr/system_int_test/ingest/subscription_ingest_test.clj

```diff
@@ -198,7 +206,7 @@
                     {:accept-format :json
                      :raw? true})
           {:keys [errors]} (ingest/parse-ingest-body :json response)]
-      (is (re-find #"required key \[Name\] not found" (first errors)))))
+      (is (re-find #"You do not have permission to perform that action." (first errors)))))
```

Comment:
This test needs to be updated. It is not testing what it is supposed to test.

Reply:
I disagree here - the point of this test is to make sure the error response comes back in the format specified in the accept header. Due to new logic, we now error our earlier than before (with a new message), but it is still coming back in the desired format.

Reply:
ignore my comment. I think the test is just testing any error in response. Even though the error message in this case is awful, I guess it is OK.

Reply:
We should probably return a more reasonable error message when the request body is empty though. Return a permission error is really confusing.

Reply:
We should either fix this test in this PR or file a ticket to get a more meaningful error message in this case.

Reply:
If a token was supplied with the request, then a the other error message would be returned - the schema validation doesn't occur until after permission ingest checks, though.

Reply:
ok, let's add a test for empty body with token for completeness.

Reply:
We should just delete these two tests that you modified. They are unnecessary if just checking for any error message because we have plenty tests that tests on specific error messages.

Reply:
nm. These are for testing accept headers. keep them.

Reply:
Agreed, two new assertions added.

Reply:
Man, it's hard to review tests in an excerpt. :)